packages/modules/pricing/src/services/pricing-module.ts

```diff
@@ -965,38 +942,29 @@ export default class PricingModuleService
     data: PricingTypes.UpdatePriceListPricesDTO[],
     sharedContext: Context = {}
   ): Promise<Price[]> {
-    const priceListIds: string[] = []
-    const priceIds: string[] = []
-
-    for (const priceListData of data) {
-      priceListIds.push(priceListData.price_list_id)
-
-      for (const price of priceListData.prices) {
-        priceIds.push(price.id)
-      }
-    }
-
-    const prices = await this.listPrices(
-      { id: priceIds },
-      { take: null, relations: ["price_rules"] },
-      sharedContext
-    )
-
-    const priceMap: Map<string, PricingTypes.PriceDTO> = new Map(
-      prices.map((price) => [price.id, price])
-    )
-
     const priceLists = await this.listPriceLists(
-      { id: priceListIds },
-      { take: null },
+      { id: data.map((p) => p.price_list_id) },
+      { take: null, relations: ["prices", "prices.price_rules"] },
       sharedContext
     )
 
-    const priceListMap = new Map(priceLists.map((p) => [p.id, p]))
+    const existingPrices = priceLists
+      .map((p) => p.prices ?? [])
+      .flat() as PricingTypes.PriceDTO[]
+
+    const pricesToCreate = data
+      .map((addPrice) =>
+        this.normalizePrices(
+          addPrice.prices as any,
```

Comment:
The typings can be improve a lot more, but I decided not to do it as part of this PR